src/application/look-up-wallet-address/app.tsx

```diff
@@ -0,0 +1,121 @@
+import { FormEvent, useEffect, useState } from 'react';
+import { useDebounce } from 'react-use';
+
+import { useCommandQuery } from 'shared/messaging';
+import { Closable, IconButton } from 'shared/ui';
+import { useExtensionSettings } from 'shared/extension';
+
+import { GetResolvedAddressCommand, GetTwitterIdsCommand } from './commands';
+import { AddressList } from './address-list';
+
+export const App = () => {
+  const [username, setUsername] = useState<string>('');
+  const [inputValue, setInputValue] = useState<string>('');
+  const [visible, setVisible] = useState(false);
+
+  const { extensionSettings } = useExtensionSettings();
+
+  const isEnabled =
+    extensionSettings['entire-extension-enabled'] &&
+    extensionSettings['block-explorers-enabled'];
+
+  const twitterIdsQuery = useCommandQuery({
+    command: new GetTwitterIdsCommand({
+      username,
+    }),
+    staleTime: Number.POSITIVE_INFINITY,
+  });
+
+  const addressesQuery = useCommandQuery({
+    command: new GetResolvedAddressCommand({
+      identifier: username,
+      twitterId: twitterIdsQuery.data?.twitterIDs
+        ? Object.values(twitterIdsQuery.data?.twitterIDs)[0]
+        : undefined,
+    }),
+    staleTime: Number.POSITIVE_INFINITY,
+    enabled: username.length >= 3,
+  });
+
+  useDebounce(
+    () => {
+      setUsername(inputValue);
+    },
+    500,
+    [inputValue],
+  );
+
+  const handleInput = (event: FormEvent<HTMLInputElement>) => {
+    const eventTarget = event.target as HTMLInputElement;
+    setInputValue(eventTarget.value);
+  };
```

Comment:
we can use react-hook-form + `watch` to get reactive value of input, I am also wondering if this debounce could not be replaced with <form> submit which submits on enter, again it's internal tool so we maybe we don't need that debounce?
CC @lennardevertz @geoist1

Reply:
my suggestion was to not call it before user presses enter/button

Reply:
fixed, now it will can the API after the "Search" button click.
![image](https://github.com/user-attachments/assets/ac262fb5-5f9c-4774-8db3-296373834878)
![image](https://github.com/user-attachments/assets/a724aa05-aa9c-4417-9111-c28d9e876009)

Reply:
nice!